### PR TITLE
[REVIEW] FEA cusort keys only

### LINKED ISSUE #381: [FEA] cusort - update snmg sort to support a keys only sort.
The SNMG implementation of sort currently only works on key/value pairs.  We should also be able to do SNMG sorting on keys only.

Should be a relatively straightforward adaptation of the existing code, just need to skip over the steps dealing with values.

### PR DESCRIPTION
**Pull request by ChuckHastings**
This PR adds a key only implementation for the SNMG sort.  Closes #381

### CODE REVIEW

File: cpp/src/sort/binning.cuh [hunk 1]
```diff
@@ -231,3 +231,120 @@ __global__ void partitionRelabel(Key_t *array,
...
+                                     counter[0][tid] + counter[1][tid]);
+  }
+
+  if (tid == 0) {
+    prefix[0] = 0L;
+    for (int p = 0 ; p < numPartitions ; ++p) {
+      prefix[p+1] = prefix[p] + counter[0][p] + counter[1][p];
+
+      if (prefix[p+1] > THREADS)
+        printf("p = %d, prefix = (%lld, %lld), counter = (%lld, %lld)\n",
```
Comment (kaatish) [inline cpp/src/sort/binning.cuh]: should there be a check for verbose flag or are we supposed to output the prefix in all the cases?

File: cpp/src/sort/sort_impl.cuh [hunk 2]
```diff
@@ -363,31 +377,54 @@ namespace cusort {
...
-      cub::DeviceRadixSort::SortPairs<Key_t,Value_t>(nullptr, tData[cpu_tid].cubSortBufferSize,
-                                                     nullptr, nullptr, nullptr, nullptr, elements);
 
-      tData[cpu_tid].bdReorder.allocate(h_writePositionsTransposed[cpu_tid][num_gpus], tData[cpu_tid].cubSortBufferSize);
+      if (keys_only) {
+        cub::DeviceRadixSort::SortKeys<Key_t>(nullptr, tData[cpu_tid].cubSortBufferSize,
+                                              nullptr, nullptr, elements);
+
+        tData[cpu_tid].bdReorder.allocate_keys_only(h_writePositionsTransposed[cpu_tid][num_gpus], tData[cpu_tid].cubSortBufferSize);
+        //tData[cpu_tid].bdReorder.allocate(h_writePositionsTransposed[cpu_tid][num_gpus], tData[cpu_tid].cubSortBufferSize);
```
Comment (kaatish) [inline cpp/src/sort/sort_impl.cuh]: commented out code can be removed.

File: cpp/src/tests/sort/sort_test.cu [hunk 3]
```diff
@@ -172,8 +205,73 @@ void verify_sorted_order(Key_t **d_key, Value_t **d_value,
...
+                              return 1;
+                            }
+                          }
+                          return 0;
+                        });
+
+      cudaDeviceSynchronize();
+      cudaCheckError();
+
+      int result = thrust::reduce(thrust::device, diffCounter, diffCounter + length, 0);
```
Comment (kaatish) [inline cpp/src/tests/sort/sort_test.cu]: Can rmm thrust policy be used here?

File: cpp/src/tests/sort/sort_test.cu [hunk 4]
```diff
@@ -172,8 +205,73 @@ void verify_sorted_order(Key_t **d_key, Value_t **d_value,
...
+
+    if (length > 0) {
+      int* diffCounter;
+      EXPECT_EQ(RMM_ALLOC(&diffCounter, sizeof(int) * length, stream), RMM_SUCCESS);
+
+      int cpu_tid = omp_get_thread_num();
+
+      Key_t *key = d_key[cpu_tid];
+
+      thrust::transform(thrust::device,
```
Comment (kaatish) [inline cpp/src/tests/sort/sort_test.cu]: can we use rmm thrust policy here?

File: cpp/src/sort/binning.cuh (hunk 1 above)
Comment (ChuckHastings) [reply cpp/src/sort/binning.cuh]: That's an old debugging thing I missed removing.

File: cpp/src/sort/sort_impl.cuh (hunk 2 above)
Comment (ChuckHastings) [reply cpp/src/sort/sort_impl.cuh]: Done

File: cpp/src/tests/sort/sort_test.cu (hunk 3 above)
Comment (ChuckHastings) [reply cpp/src/tests/sort/sort_test.cu]: Updated thrust::device references.

**Pull request merged**